lib/shim/shim.js

```diff
@@ -2100,3 +2101,29 @@ function _es5WrapClass(shim, Base, fnName, spec, args) {
 
   return WrappedClass
 }
+
+/**
+ * Method for prefixing Route (aka URL) parameters with `request.parameters.route`
+ *
+ * Many web frameworks support adding parameters to routes when defining your API structure, and this function
+ * updates those parameters names to be prefixed by `request.parameters.route`. This is to avoid collision with reserved
+ * attribute names, as parameters used to be blindly stored on router span attributes (see https://github.com/newrelic/node-newrelic/issues/1574)
+ * in addition to being prefixed by `request.parameters`.
+ *
+ * Route parameters used to be stored under `request.parameters.*` just like query parameters pre v10, but we
+ * now prefix with `request.parameter.route` to avoid collision in the event an application uses the same name for a query and route
+ * parameter. Additionally, we now store the same key on the attributes of the base segment, trace, and router span.
+ *
+ * Exported on shim to be used in our Next.js instrumentation, as that instrumentation does not follow the same pattern as all the other
+ * web frameworks we support.
+ *
+ * @param {object} params - Object containing route/url parameter key/value pairs
+ * @returns {object} the updated object, `key` will now be `request.parameters.route.key`, value remains untouched
+ */
+function prefixRouteParameters(params) {
```

Comment:
I think you should add some unit tests on this since it's a public method.  I know it's getting indirectly tested but always good to find tests for methods directly too